console-subscriber/src/aggregator.rs

```diff
@@ -67,15 +82,56 @@ pub(crate) struct Flush {
     pub(crate) triggered: AtomicBool,
 }
 
-struct Stats {
-    // task stats
-    polls: u64,
+// An entity that at some point in time can be closed
+// and recycled
+trait Closable {
+    fn closed_at(&self) -> &Option<SystemTime>;
+}
+
+trait ToProto {
+    type Result;
+    fn to_proto(&self) -> Self::Result;
+}
+struct PollStats {
```

Comment:
nit: prefer whitespace between definitions
```suggestion
}

struct PollStats {
```